<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="Exception and warning types defined by Python."><title>pyo3::exceptions - Rust</title><script>if(window.location.protocol!=="file:")document.head.insertAdjacentHTML("beforeend","SourceSerif4-Regular-6b053e98.ttf.woff2,FiraSans-Regular-0fe48ade.woff2,FiraSans-Medium-e1aa3f0a.woff2,SourceCodePro-Regular-8badfe75.ttf.woff2,SourceCodePro-Semibold-aa29a496.ttf.woff2".split(",").map(f=>`<link rel="preload" as="font" type="font/woff2" crossorigin href="../../static.files/${f}">`).join(""))</script><link rel="stylesheet" href="../../static.files/normalize-9960930a.css"><link rel="stylesheet" href="../../static.files/rustdoc-46132b98.css"><meta name="rustdoc-vars" data-root-path="../../" data-static-root-path="../../static.files/" data-current-crate="pyo3" data-themes="" data-resource-suffix="" data-rustdoc-version="1.85.1 (4eb161250 2025-03-15)" data-channel="1.85.1" data-search-js="search-75f5ac3e.js" data-settings-js="settings-0f613d39.js" ><script src="../../static.files/storage-59e33391.js"></script><script defer src="../sidebar-items.js"></script><script defer src="../../static.files/main-5f194d8c.js"></script><noscript><link rel="stylesheet" href="../../static.files/noscript-893ab5e7.css"></noscript><link rel="alternate icon" type="image/png" href="../../static.files/favicon-32x32-6580c154.png"><link rel="icon" type="image/svg+xml" href="../../static.files/favicon-044be391.svg"></head><body class="rustdoc mod"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle" title="show sidebar"></button></nav><nav class="sidebar"><div class="sidebar-crate"><h2><a href="../../pyo3/index.html">pyo3</a><span class="version">0.24.1</span></h2></div><div class="sidebar-elems"><section id="rustdoc-toc"><h2 class="location"><a href="#">Module exceptions</a></h2><h3><a href="#modules">Module Items</a></h3><ul class="block"><li><a href="#modules" title="Modules">Modules</a></li><li><a href="#structs" title="Structs">Structs</a></li></ul></section><div id="rustdoc-modnav"><h2 class="in-crate"><a href="../index.html">In crate pyo3</a></h2></div></div></nav><div class="sidebar-resizer"></div><main><div class="width-limiter"><rustdoc-search></rustdoc-search><section id="main-content" class="content"><div class="main-heading"><span class="rustdoc-breadcrumbs"><a href="../index.html">pyo3</a></span><h1>Module <span>exceptions</span><button id="copy-path" title="Copy item path to clipboard">Copy item path</button></h1><rustdoc-toolbar></rustdoc-toolbar><span class="sub-heading"><a class="src" href="../../src/pyo3/exceptions.rs.html#1-1152">Source</a> </span></div><details class="toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>Exception and warning types defined by Python.</p>
<p>The structs in this module represent Python’s built-in exceptions and
warnings, while the modules comprise structs representing errors defined in
Python code.</p>
<p>The latter are created with the
<a href="../macro.import_exception.html" title="macro pyo3::import_exception"><code>import_exception</code></a> macro, which you can use
yourself to import Python classes that are ultimately derived from
<code>BaseException</code>.</p>
</div></details><h2 id="modules" class="section-header">Modules<a href="#modules" class="anchor">§</a></h2><ul class="item-table"><li><div class="item-name"><a class="mod" href="asyncio/index.html" title="mod pyo3::exceptions::asyncio">asyncio</a></div><div class="desc docblock-short">Exceptions defined in Python’s <a href="https://docs.python.org/3/library/asyncio.html"><code>asyncio</code></a>
module.</div></li><li><div class="item-name"><a class="mod" href="socket/index.html" title="mod pyo3::exceptions::socket">socket</a></div><div class="desc docblock-short">Exceptions defined in Python’s <a href="https://docs.python.org/3/library/socket.html"><code>socket</code></a>
module.</div></li></ul><h2 id="structs" class="section-header">Structs<a href="#structs" class="anchor">§</a></h2><ul class="item-table"><li><div class="item-name"><a class="struct" href="struct.PyArithmeticError.html" title="struct pyo3::exceptions::PyArithmeticError">PyArithmetic<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#ArithmeticError"><code>ArithmeticError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyAssertionError.html" title="struct pyo3::exceptions::PyAssertionError">PyAssertion<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#AssertionError"><code>AssertionError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyAttributeError.html" title="struct pyo3::exceptions::PyAttributeError">PyAttribute<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#AttributeError"><code>AttributeError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyBaseException.html" title="struct pyo3::exceptions::PyBaseException">PyBase<wbr>Exception</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#BaseException"><code>BaseException</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyBaseExceptionGroup.html" title="struct pyo3::exceptions::PyBaseExceptionGroup">PyBase<wbr>Exception<wbr>Group</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#BaseExceptionGroup"><code>BaseExceptionGroup</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyBlockingIOError.html" title="struct pyo3::exceptions::PyBlockingIOError">PyBlockingIO<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#BlockingIOError"><code>BlockingIOError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyBrokenPipeError.html" title="struct pyo3::exceptions::PyBrokenPipeError">PyBroken<wbr>Pipe<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#BrokenPipeError"><code>BrokenPipeError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyBufferError.html" title="struct pyo3::exceptions::PyBufferError">PyBuffer<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#BufferError"><code>BufferError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyBytesWarning.html" title="struct pyo3::exceptions::PyBytesWarning">PyBytes<wbr>Warning</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#BytesWarning"><code>BytesWarning</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyChildProcessError.html" title="struct pyo3::exceptions::PyChildProcessError">PyChild<wbr>Process<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#ChildProcessError"><code>ChildProcessError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyConnectionAbortedError.html" title="struct pyo3::exceptions::PyConnectionAbortedError">PyConnection<wbr>Aborted<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#ConnectionAbortedError"><code>ConnectionAbortedError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyConnectionError.html" title="struct pyo3::exceptions::PyConnectionError">PyConnection<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#ConnectionError"><code>ConnectionError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyConnectionRefusedError.html" title="struct pyo3::exceptions::PyConnectionRefusedError">PyConnection<wbr>Refused<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#ConnectionRefusedError"><code>ConnectionRefusedError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyConnectionResetError.html" title="struct pyo3::exceptions::PyConnectionResetError">PyConnection<wbr>Reset<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#ConnectionResetError"><code>ConnectionResetError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyDeprecationWarning.html" title="struct pyo3::exceptions::PyDeprecationWarning">PyDeprecation<wbr>Warning</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#DeprecationWarning"><code>DeprecationWarning</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyEOFError.html" title="struct pyo3::exceptions::PyEOFError">PyEOF<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#EOFError"><code>EOFError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyEncodingWarning.html" title="struct pyo3::exceptions::PyEncodingWarning">PyEncoding<wbr>Warning</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#EncodingWarning"><code>EncodingWarning</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyEnvironmentError.html" title="struct pyo3::exceptions::PyEnvironmentError">PyEnvironment<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#EnvironmentError"><code>EnvironmentError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyException.html" title="struct pyo3::exceptions::PyException">PyException</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#Exception"><code>Exception</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyFileExistsError.html" title="struct pyo3::exceptions::PyFileExistsError">PyFile<wbr>Exists<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#FileExistsError"><code>FileExistsError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyFileNotFoundError.html" title="struct pyo3::exceptions::PyFileNotFoundError">PyFile<wbr>NotFound<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#FileNotFoundError"><code>FileNotFoundError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyFloatingPointError.html" title="struct pyo3::exceptions::PyFloatingPointError">PyFloating<wbr>Point<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#FloatingPointError"><code>FloatingPointError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyFutureWarning.html" title="struct pyo3::exceptions::PyFutureWarning">PyFuture<wbr>Warning</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#FutureWarning"><code>FutureWarning</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyGeneratorExit.html" title="struct pyo3::exceptions::PyGeneratorExit">PyGenerator<wbr>Exit</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#GeneratorExit"><code>GeneratorExit</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyIOError.html" title="struct pyo3::exceptions::PyIOError">PyIO<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#IOError"><code>IOError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyImportError.html" title="struct pyo3::exceptions::PyImportError">PyImport<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#ImportError"><code>ImportError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyImportWarning.html" title="struct pyo3::exceptions::PyImportWarning">PyImport<wbr>Warning</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#ImportWarning"><code>ImportWarning</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyIndexError.html" title="struct pyo3::exceptions::PyIndexError">PyIndex<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#IndexError"><code>IndexError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyInterruptedError.html" title="struct pyo3::exceptions::PyInterruptedError">PyInterrupted<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#InterruptedError"><code>InterruptedError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyIsADirectoryError.html" title="struct pyo3::exceptions::PyIsADirectoryError">PyIsA<wbr>Directory<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#IsADirectoryError"><code>IsADirectoryError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyKeyError.html" title="struct pyo3::exceptions::PyKeyError">PyKey<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#KeyError"><code>KeyError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyKeyboardInterrupt.html" title="struct pyo3::exceptions::PyKeyboardInterrupt">PyKeyboard<wbr>Interrupt</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#KeyboardInterrupt"><code>KeyboardInterrupt</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyLookupError.html" title="struct pyo3::exceptions::PyLookupError">PyLookup<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#LookupError"><code>LookupError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyMemoryError.html" title="struct pyo3::exceptions::PyMemoryError">PyMemory<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#MemoryError"><code>MemoryError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyModuleNotFoundError.html" title="struct pyo3::exceptions::PyModuleNotFoundError">PyModule<wbr>NotFound<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#ModuleNotFoundError"><code>ModuleNotFoundError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyNameError.html" title="struct pyo3::exceptions::PyNameError">PyName<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#NameError"><code>NameError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyNotADirectoryError.html" title="struct pyo3::exceptions::PyNotADirectoryError">PyNotA<wbr>Directory<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#NotADirectoryError"><code>NotADirectoryError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyNotImplementedError.html" title="struct pyo3::exceptions::PyNotImplementedError">PyNot<wbr>Implemented<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#NotImplementedError"><code>NotImplementedError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyOSError.html" title="struct pyo3::exceptions::PyOSError">PyOS<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#OSError"><code>OSError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyOverflowError.html" title="struct pyo3::exceptions::PyOverflowError">PyOverflow<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#OverflowError"><code>OverflowError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyPendingDeprecationWarning.html" title="struct pyo3::exceptions::PyPendingDeprecationWarning">PyPending<wbr>Deprecation<wbr>Warning</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#PendingDeprecationWarning"><code>PendingDeprecationWarning</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyPermissionError.html" title="struct pyo3::exceptions::PyPermissionError">PyPermission<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#PermissionError"><code>PermissionError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyProcessLookupError.html" title="struct pyo3::exceptions::PyProcessLookupError">PyProcess<wbr>Lookup<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#ProcessLookupError"><code>ProcessLookupError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyRecursionError.html" title="struct pyo3::exceptions::PyRecursionError">PyRecursion<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#RecursionError"><code>RecursionError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyReferenceError.html" title="struct pyo3::exceptions::PyReferenceError">PyReference<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#ReferenceError"><code>ReferenceError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyResourceWarning.html" title="struct pyo3::exceptions::PyResourceWarning">PyResource<wbr>Warning</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#ResourceWarning"><code>ResourceWarning</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyRuntimeError.html" title="struct pyo3::exceptions::PyRuntimeError">PyRuntime<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#RuntimeError"><code>RuntimeError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyRuntimeWarning.html" title="struct pyo3::exceptions::PyRuntimeWarning">PyRuntime<wbr>Warning</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#RuntimeWarning"><code>RuntimeWarning</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyStopAsyncIteration.html" title="struct pyo3::exceptions::PyStopAsyncIteration">PyStop<wbr>Async<wbr>Iteration</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#StopAsyncIteration"><code>StopAsyncIteration</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyStopIteration.html" title="struct pyo3::exceptions::PyStopIteration">PyStop<wbr>Iteration</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#StopIteration"><code>StopIteration</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PySyntaxError.html" title="struct pyo3::exceptions::PySyntaxError">PySyntax<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#SyntaxError"><code>SyntaxError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PySyntaxWarning.html" title="struct pyo3::exceptions::PySyntaxWarning">PySyntax<wbr>Warning</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#SyntaxWarning"><code>SyntaxWarning</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PySystemError.html" title="struct pyo3::exceptions::PySystemError">PySystem<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#SystemError"><code>SystemError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PySystemExit.html" title="struct pyo3::exceptions::PySystemExit">PySystem<wbr>Exit</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#SystemExit"><code>SystemExit</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyTimeoutError.html" title="struct pyo3::exceptions::PyTimeoutError">PyTimeout<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#TimeoutError"><code>TimeoutError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyTypeError.html" title="struct pyo3::exceptions::PyTypeError">PyType<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#TypeError"><code>TypeError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyUnboundLocalError.html" title="struct pyo3::exceptions::PyUnboundLocalError">PyUnbound<wbr>Local<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#UnboundLocalError"><code>UnboundLocalError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyUnicodeDecodeError.html" title="struct pyo3::exceptions::PyUnicodeDecodeError">PyUnicode<wbr>Decode<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#UnicodeDecodeError"><code>UnicodeDecodeError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyUnicodeEncodeError.html" title="struct pyo3::exceptions::PyUnicodeEncodeError">PyUnicode<wbr>Encode<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#UnicodeEncodeError"><code>UnicodeEncodeError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyUnicodeError.html" title="struct pyo3::exceptions::PyUnicodeError">PyUnicode<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#UnicodeError"><code>UnicodeError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyUnicodeTranslateError.html" title="struct pyo3::exceptions::PyUnicodeTranslateError">PyUnicode<wbr>Translate<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#UnicodeTranslateError"><code>UnicodeTranslateError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyUnicodeWarning.html" title="struct pyo3::exceptions::PyUnicodeWarning">PyUnicode<wbr>Warning</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#UnicodeWarning"><code>UnicodeWarning</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyUserWarning.html" title="struct pyo3::exceptions::PyUserWarning">PyUser<wbr>Warning</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#UserWarning"><code>UserWarning</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyValueError.html" title="struct pyo3::exceptions::PyValueError">PyValue<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#ValueError"><code>ValueError</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyWarning.html" title="struct pyo3::exceptions::PyWarning">PyWarning</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#Warning"><code>Warning</code></a> exception.</div></li><li><div class="item-name"><a class="struct" href="struct.PyZeroDivisionError.html" title="struct pyo3::exceptions::PyZeroDivisionError">PyZero<wbr>Division<wbr>Error</a></div><div class="desc docblock-short">Represents Python’s <a href="https://docs.python.org/3/library/exceptions.html#ZeroDivisionError"><code>ZeroDivisionError</code></a> exception.</div></li></ul></section></div></main></body></html>